README.lhs

```diff
@@ -30,53 +51,30 @@ main = someExample
 ```
 -->
 
-## Development & Tests
-
-```console
-stack build --fast --pedantic --test --file-watch
-```
-
----
-
-## How to use this Template
+## Gotchas
 
-Haskell library template used at Freckle.
+There are cases where it's not practical to match the aeson functionality
+exactly when we're making compatibility functions against older (or newer)
+versions. The following lists concessions we've made.
 
-### Create your repo
+- Our version of `alterF` (`unordered-containers < 0.2.10`) may compute the hash
+  key multiple times, since we don't have access to the internals necessary to
+  pre-compute the key once and use it for any lookup, insert, and delete
+  actions.
+- We may use strict implementations for should-be lazy functions (or vice
+  versa). For example, in `unordered-containers < 0.2.11`, `foldl = foldl'` and
+  `foldr' = foldr`.
+- Only one of `coercionToHashMap` or `coercionToMap` will be `Just`, depending
+  on your actual aeson version.
+- Either conversions dealing with `HashMap` or those dealing with `Map` will be
+  fast (runtime-erased), depending on your actual aeson version.
 
-```sh
-gh repo create --template freckle/haskell-library-template --public freckle/<name>
-git clone git@github.com:freckle/<name>
-cd ./<name>
-```
-
-### Rename your package
-
-```sh
-find -type f -exec \
-  sed -i s/haskell-library-template/my-name/ {} +
-```
-
-Edit `package.yaml` as necessary.
-
-### Enable release
-
-When you are ready to release your library, simply remove the conditional from
-the release workflow.
+## Development & Tests
 
-```diff
--      - if: false # Remove when ready to release
+```console
+stack build --fast --pedantic --test --file-watch
```

Comment:
```suggestion
stack build --fast --test --file-watch
```

Long story, but `--pedantic` won't work because we're disabling `unused-imports` in this repository. Since that's part of `-Wall`, which `--pedantic` adds after our `-Weverthing -Wno-*` stanzas, it gets re-enabled.